packages/-ember-data/tests/integration/record-data/store-wrapper-test.ts

```diff
@@ -0,0 +1,190 @@
+import { get } from '@ember/object';
+import { setupTest } from 'ember-qunit';
+import Model from 'ember-data/model';
+import Store from 'ember-data/store';
+import { attr, belongsTo, hasMany } from '@ember-decorators/data';
+import { module, test } from 'qunit';
+import { settled } from '@ember/test-helpers';
+import publicArgs from '../../helpers/public-props';
+
+class Person extends Model {
+  @attr('string', {})
+  name;
+}
+
+class Car extends Model {
+  @belongsTo('house')
+  garage;
+
+  @attr('string', {})
+  make;
+}
+
+class House extends Model {
+  @attr('string', {})
+  name;
+
+  @belongsTo('person', { async: false })
+  landlord;
+
+  @belongsTo('car', { async: false })
+  car;
+
+  @hasMany('person', { async: false })
+  tenants;
+}
+
+// TODO Make the class implement the RD interface
+class TestRecordData {
+  pushData(data, calculateChange?: boolean) { }
+  clientDidCreate() { }
+
+  willCommit() { }
+
+  commitWasRejected() { }
+
+  unloadRecord() { }
+  rollbackAttributes() { }
+  changedAttributes(): any { }
+
+  hasChangedAttributes(): boolean {
+    return false;
+  }
+
+  setDirtyAttribute(key: string, value: any) { }
+
+  getAttr(key: string): string {
+    return "test";
+  }
+
+  hasAttr(key: string): boolean {
+    return false;
+  }
+
+  getHasMany(key: string) {
+    return {};
+  }
+
+  addToHasMany(key: string, recordDatas: this[], idx?: number) { }
+  removeFromHasMany(key: string, recordDatas: this[]) { }
+  setDirtyHasMany(key: string, recordDatas: this[]) { }
+
+  getBelongsTo(key: string) { }
+
+  setDirtyBelongsTo(name: string, recordData: this | null) { }
+
+  didCommit(data) { }
+
+  isAttrDirty(key: string) { return false; }
+  removeFromInverseRelationships(isNew: boolean) { }
+
+  _initRecordCreateOptions(options) { }
+}
+
+let CustomStore = Store.extend({
+  createRecordDataFor(modelName, id, clientId, storeWrapper) {
+    return new TestRecordData();
+  }
+});
+
+
+module('integration/store-wrapper - RecordData StoreWrapper tests', function (hooks) {
+  setupTest(hooks);
+
+  let store;
+
+  hooks.beforeEach(function () {
+    let { owner } = this;
+
+    owner.register('model:person', Person);
+    owner.register('model:house', House);
+    owner.register('model:car', Car);
+    owner.register('service:store', CustomStore);
+  });
+
+  test("Relationship definitions", async function (assert) {
```

Comment:
this test looks good; we should probably add tests for the rest of the store wrapper api though no?

Reply:
added